x-pack/plugins/viz_editor/public/common/components/operation_editor/operation_definitions.tsx

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFieldNumber, EuiForm, EuiFormRow, EuiIcon, EuiRange, EuiSelect } from '@elastic/eui';
+import React from 'react';
+import {
+  AvgOperation,
+  CardinalityOperation,
+  ColumnOperation,
+  CountOperation,
+  DatasourceField,
+  DateHistogramOperation,
+  FieldOperation,
+  isFieldApplicableForScale,
+  SelectOperation,
+  SelectOperator,
+  SumOperation,
+  TermsOperation,
+  WindowFunction,
+  WindowOperation,
+} from '../../../../common';
+import { operationToName, VisModel } from '../../lib';
+
+const FixedEuiRange = EuiRange as React.ComponentType<any>;
+
+export type Scale = 'ordinal' | 'interval';
+export type Cardinality = 'single' | 'multi';
+
+export interface OperationEditorProps {
+  children: any;
+  column: SelectOperation;
+  onColumnChange: (newColumn: SelectOperation) => void;
+  onColumnRemove?: () => void;
+  removable?: boolean;
+  visModel: VisModel;
+  allowedScale: Scale;
+  allowedCardinality: Cardinality;
+  defaultOperator: (field: DatasourceField) => SelectOperator;
+  canDrop?: (field: DatasourceField) => boolean;
+}
+
+export interface OperationDefinition {
+  // The human-friendly name of the operation
+  name: string;
+
+  // The type of the operation (e.g. 'sum', 'avg', etc)
+  operator: SelectOperation['operator'];
+
+  // Filter the fields list down to only those supported by this
+  // operation (e.g. numbers for sum operations, dates for histograms)
+  applicableFields: (fields: DatasourceField[], props: OperationEditorProps) => DatasourceField[];
+
+  // Provide a textual summary of the operation, maybe should return
+  // a React component instead of a string?
+  summarize?: (operation: any) => React.ReactNode;
+
+  // Convert the specified current operation into a different operation
+  // e.g. from count to sum, etc.
+  toSelectClause: (
```

Comment:
What is a "select clause"? Do you mean `toSelectOperation(`?

Reply:
Good question. I suspect it was meant to return the complete select operation array, but since it doesn't at the moment, I changed it to `toSelectOperation`